### ISSUE TITLE
Issues with XamlBehaviors and implemented system events

### ISSUE THREAD

**Issue by bigtlb** [MyToolkit/MyToolkit#44]
I have opened an [issue over with the XamlBehaviors team][1] about this as well.  I don't know who should be responsible.  But, I wanted to make you were aware as well.

The new XamlBehaviors have some custom logic in the EventTriggerBehavior for Windows Runtime (UWP) apps.  If the event handler delegate is defined in an assembly from the Windows Runtime, they assume the even will be handled in the runtime as well and will require marshalling.

#### XamlBehavior EventTriggerBehavior.cs Code
``` csharp
this.isWindowsRuntimeEvent = EventTriggerBehavior.IsWindowsRuntimeType(info.EventHandlerType);
if (this.isWindowsRuntimeEvent)
{
    this.addEventHandlerMethod = add => (EventRegistrationToken)info.AddMethod.Invoke(this.resolvedSource, new object[] { add });
    this.removeEventHandlerMethod = token => info.RemoveMethod.Invoke(this.resolvedSource, new object[] { token });

    WindowsRuntimeMarshal.AddEventHandler(this.addEventHandlerMethod, this.removeEventHandlerMethod, this.eventHandler);
}
else
{
    info.AddEventHandler(this.resolvedSource, this.eventHandler);
}
```

This causes issues in the DataGrid which is reusing the SelectionChangedEventHandler delegate definition.

I was able to work around it easily in my application by copying and pasting the delegate definition into a copy of your source code.  But, I  am not sure if that is the most elegant solution, or how many other instances like that exist.

``` csharp
    /// <summary>A data grid control. </summary>
    public class DataGrid : Control
    {
        private Grid _titleRowControl;
        private MtListBox _listControl;
        private bool _initialized = false;
        private object _initialSelectedItem;
        private object _initialFilter;

        public delegate void SelectionChangedEventHandler(System.Object sender, SelectionChangedEventArgs e);
```

[1]: https://github.com/Microsoft/XamlBehaviors/issues/76

**Comment by rsuter** [comment]
I think the best option is to reimplement the delegate... waiting for the MS team to fix the issue will probably take longer...

**Comment by bigtlb** [comment]
I haven't tested to see how pervasive this issue is.  For instance, if this also affects controls with EventHandler<T> declarations.

**Comment by rsuter** [comment]
Fixed in v2.5.11
